drivers/wiegand/wiegand.c

```diff
@@ -0,0 +1,329 @@
+/*
+ * Copyright (C) 2019 Hackerspace San Salvador
+ *
+ * This driver is a port for RIOT-OS based on the Wiegand protocol for
+ * Arduino implementation from monkeyboards.org.
+ *
+ * Original implementation available at:
+ *   https://github.com/monkeyboard/Wiegand-Protocol-Library-for-Arduino
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ *
+ */
+
+/**
+ * @ingroup     drivers_wiegand
+ * @{
+ *
+ * @file
+ * @brief       Driver for the Wiegand Protocol.
+ *
+ * @author      Mario Gomez <mario.gomez@teubi.co>
+ *
+ * @}
+ */
+
+#include "xtimer.h"
+#include "wiegand_params.h"
+
+/**
+ * @brief Handler for D0 signal in Wiegand protocol
+ *
+ */
+static void wiegand_read_d0(void *arg)
+{
+    wiegand_t *dev = (wiegand_t *)arg;
+
+    // Increment bit count for Interrupt connected to D0
+    dev->wg_bit_count++;
+    // If bit count more than 31, process high bits
+    if (dev->wg_bit_count > 31) {
+        // shift value to high bits
+        dev->wg_card_temp_high |= ((0x80000000 & dev->wg_card_temp) >> 31);
+        dev->wg_card_temp_high <<= 1;
+        dev->wg_card_temp <<= 1;
```

Comment:
You could also use `uint64_t wg_card_temp`, then you don't have to do this distinction.
I wonder how it affects code size though.